cyber/scheduler/policy/classic_context.cc

```diff
@@ -33,20 +33,19 @@ namespace {
 static constexpr auto MIN_SLEEP_INTERVAL = std::chrono::milliseconds(1);
 }
 
-std::mutex ClassicContext::mtx_wq_;
-std::condition_variable ClassicContext::cv_wq_;
-std::array<AtomicRWLock, MAX_PRIO> ClassicContext::rq_locks_;
-std::array<std::vector<std::shared_ptr<CRoutine>>, MAX_PRIO>
-    ClassicContext::rq_;
+GRP_WQ_MUTEX ClassicContext::mtx_wq_;
+GRP_WQ_CV ClassicContext::cv_wq_;
+RQ_LOCK_GROUP ClassicContext::rq_locks_;
+CR_GROUP ClassicContext::cr_group_;
 
 std::shared_ptr<CRoutine> ClassicContext::NextRoutine() {
   if (unlikely(stop_)) {
     return nullptr;
   }
 
   for (int i = MAX_PRIO - 1; i >= 0; --i) {
-    ReadLockGuard<AtomicRWLock> lk(rq_locks_[i]);
-    for (auto& cr : rq_[i]) {
+    ReadLockGuard<AtomicRWLock> lk(rq_locks_[group_name_].at(i));
```

Comment:
advise assign rq_locks_[group_name_] to a local variable to avoid search the hash map every time when fetch next croutine